### DISCUSSION TITLE
Help with v0.14.0-beta.1 on MacOS 13 - VM hanging on requirement 2 of 3 / `sudo: parse error in /etc/sudoers.d/90-cloud-init-users near line 14`

### DISCUSSION THREAD

**Discussion by thstraus** [lima-vm/lima#1218] [Q&A]
Hi,

not sure this is the right place to get help. Hope someone can give me hints on how to debug this:

I'm experimenting with lima version v0.14.0-beta.1 on an M1 Mac with MacOS 13. I got somehow to a point where my VM is in an unknown state. It hangs on 

[hostagent] Waiting for the essential requirement 2 of 3: "user session is ready for ssh"

Strange enough, while it is hanging on this, I CAN connect from a different terminal to the VM. But it never gets further the requirement.

Has anybody an idea on how to debug this? Maybe this might also be helpful for the developers?

Many thanks for any kind hint!

Thomas

BTW: I've got to using lima as Multipass on Ventura has left all my VMs in unknown state...

PS: Forgot to say, I'm using [VZ]. This is the reason why I'm using the beta version

**Comment by AkihiroSuda** [comment]
Some errors might be available in `/var/log/cloud-init-output.log` in the guest

**Reply by thstraus** [reply]
Many thanks for your reply. Unfortunately, the guest is in a state I can ssh into it, but sudo is not working. Most of the log files are readable only by root...
Issuing sudo results in this error:

>>> /etc/sudoers.d/90-cloud-init-users: syntax error near line 14 <<<
sudo: parse error in /etc/sudoers.d/90-cloud-init-users near line 14
sudo: no valid sudoers sources found, quitting
sudo: unable to initialize policy plugin

**Reply by AkihiroSuda** [reply]
Does your username contain a non-alphanumeric character?

**Reply by AkihiroSuda** [reply]
Also, does it work with the default qemu mode?

**Reply by thstraus** [reply]
Just to be clear, the VM was working fine, no problem at all. In some way, I probably am unable to reproduce, the working VM ended now in this unclear state. I have no problem in starting over with a fresh VM, but I think it would be important to understand what this state is and how one could recover from it. (It happened to ALL my VMs under Multipass, since I upgraded to MacOS 13.0.1 ...).

I have no strange characters in my username. But I was copying SSH keys over from host to guest. Not sure if I can brake things through this.

About qemu, do you mean I can bring this instance to start with qemu virtualisation?

**Reply by thstraus** [reply]
Is there a way to boot the guest into single user mode?

**Reply by thstraus** [reply]
Again I'm not sure this all makes sense to you, but I see a difference in a successful startup and the hanging one. A VM (based on VZ) that boots fine gives an error, while the hanging VM does not. 
The error during the working boot is the following

INFO[0016] [hostagent] Waiting for the essential requirement 2 of 3: "user session is ready for ssh" 
ERRO[0017] [hostagent] r.CreateEndpoint() = connection was refused 
INFO[0026] [hostagent] Waiting for the essential requirement 2 of 3: "user session is ready for ssh" 
INFO[0026] [hostagent] The essential requirement 2 of 3 is satisfied

This error does NOT appear on the hanging VM

**Reply by AkihiroSuda** [reply]
> single user mode

The video display can be enabled here, so maybe you can drop into the single mode via the GRUB menu (append `init=/bin/bash` to the kernel cmdline)

https://github.com/lima-vm/lima/blob/fc3f2a9a510dcbb86794c048a2b116fea9eb6741/examples/default.yaml#L231-L238
